pkg/asset/agent/manifests/infraenv.go

```diff
@@ -74,6 +74,12 @@ func (i *InfraEnv) Generate(_ context.Context, dependencies asset.Parents) error
 				i.Config.Spec.AdditionalNTPSources = agentConfig.Config.AdditionalNTPSources
 			}
 
+			if installConfig.Config.BareMetal != nil {
+				if i.Config.Spec.AdditionalNTPSources == nil && installConfig.Config.BareMetal.AdditionalNTPServers != nil {
```

Comment:
nit: In fact, you don't need the check on AdditionalNTPServers: worst case it will assign "nil" to "nil".

Reply:
What I mean is just

```go
if installConfig.Config.BareMetal != nil && i.Config.Spec.AdditionalNTPSources == nil {
    i.Config.Spec.AdditionalNTPSources = installConfig.Config.BareMetal.AdditionalNTPServers
}
```

Reply:
I intentionally did this more verbose and in nested if.  While it does the same thing I think it's a lot easier to read and understand the intent this way. My opinion.